runtime/jcl/common/java_lang_Class.cpp

```diff
@@ -1785,5 +1820,72 @@ Java_java_lang_Class_getNestHostImpl(JNIEnv *env, jobject recv)
 	vmFuncs->internalReleaseVMAccess(currentThread);
 	return result;
 }
+
+jobject JNICALL
+Java_java_lang_Class_getNestMembersImpl(JNIEnv *env, jobject recv)
+{
+	J9VMThread *currentThread = (J9VMThread*)env;
+	J9JavaVM *vm = currentThread->javaVM;
+	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+	J9MemoryManagerFunctions *mmFuncs = vm->memoryManagerFunctions;
+
+	j9object_t resultObject = NULL;
+	jobject result = NULL;
+
+	vmFuncs->internalEnterVMFromJNI(currentThread);
+
+	J9Class *clazz = J9VM_J9CLASS_FROM_HEAPCLASS(currentThread, J9_JNI_UNWRAP_REFERENCE(recv));
+	J9Class *nestHost = clazz->nestHost;
+	J9ROMClass *romHostClass = nestHost->romClass;
+	U_16 nestMemberCount = romHostClass->nestMemberCount;
+
+	/*  Grab the Class<?> class for result object size */
+	J9Class *jlClass = J9VMJAVALANGCLASS(vm);
+	J9Class *arrayClass = fetchArrayClass(currentThread, jlClass);
+	if (NULL != currentThread->currentException) {
+		goto _throwException;
+	}
+
+	resultObject = mmFuncs->J9AllocateIndexableObject(currentThread, arrayClass, 1 + nestMemberCount, J9_GC_ALLOCATE_OBJECT_NON_INSTRUMENTABLE);
+	if (NULL == resultObject) {
+		vmFuncs->setHeapOutOfMemoryError(currentThread);
+		goto _throwException;
+	}
+
+	/* Host class is always in zeroeth index */
+	J9JAVAARRAYOFOBJECT_STORE(currentThread, resultObject, 0, J9VM_J9CLASS_TO_HEAPCLASS(nestHost));
+
+	/* If host class claims nest members, they should be placed in second index onwards */
+	if (0 != nestMemberCount) {
+		J9SRP *nestMembers = J9ROMCLASS_NESTMEMBERS(romHostClass);
+		U_16 i = 0;
+		/* Classes in nest are in same runtime package & therefore have same classloader */
+		J9ClassLoader *classLoader = clazz->classLoader;
+
+		for (i = 0; i < nestMemberCount; i++) {
+			J9Class *nestMember = NULL;
+			J9UTF8 *nestMemberName = NNSRP_GET(nestMembers[i],J9UTF8*);
+
+			PUSH_OBJECT_IN_SPECIAL_FRAME(currentThread, resultObject);
+			nestMember = vmFuncs->internalFindClassUTF8(currentThread, J9UTF8_DATA(nestMemberName), J9UTF8_LENGTH(nestMemberName), classLoader, J9_FINDCLASS_FLAG_THROW_ON_FAIL);
+			resultObject = POP_OBJECT_IN_SPECIAL_FRAME(currentThread);
+
+			if (NULL == nestMember) {
+				goto _throwException;
+			} else if (nestMember->nestHost != nestHost) {
+				setNestmatesIncompatibleClassChangeError(currentThread, nestMember, nestHost);
+				goto _throwException;
+			} else {
+				J9JAVAARRAYOFOBJECT_STORE(currentThread, resultObject, i + 1, J9VM_J9CLASS_TO_HEAPCLASS(nestMember));
+			}
+		}
+	}
+
+	result = vmFuncs->j9jni_createLocalRef(env, resultObject);
+
+_throwException:
```

Comment:
This name of this label is misleading: jumping here doesn't throw an an exception. Perhaps just 'done' or 'exit'.